src/markov/mc_tools.jl

```diff
@@ -400,7 +365,7 @@ initial states.
 """
 function simulate!(mc::MarkovChain, X::Matrix{Int})
     n = size(mc.p, 1)
-    P_dist = [DiscreteRV(vec(mc.p[i, :])) for i in 1:n]
+    P_dist = [DiscreteRV(vec(full(mc.p[i, :]))) for i in 1:n]
```

Comment:
This is the one main thing I don't feel great about in this PR. Notice that I added `full` around each row of the stochastic matrix. 

`DiscreteRV` uses a `cumsum` over an `AbstractVector` to give us the transition to the next state. In Julia 0.4 there is no `SparseVector` type (only sparse matrix), so to get this to be an `AbstractVector` we needed to make it dense before passing to `DiscreteRV`. The thing that makes it ok is that `cumsum` is an operation that will make any sparse array dense, after the first non-zero element. So we probably are more efficient working directly with the dense array than we would have been working with a dense sparse array, but I just wanted to point this out.

Also, I checked the native machine code for this function with a dense `mc.p` before and after making this change and the machine code was identical. this means that when `mc.p` is already dense the jit compiler completely removes the no-op that `full` causes and we have 0 performance hit.